docs/Brew-Livecheck.md

```diff
@@ -268,6 +268,34 @@ livecheck do
 end
 ```
 
+If you need to modify the version, you can access the yaml hash in the strategy block like so:
+
+```ruby
+livecheck do
+  url "https://example.org/my-app/latest-mac.yml"
+  strategy :electron_builder do |yaml|
+    yaml["version"]&.gsub(/\D/, "")
+  end
+end
+```
+
+Or a more complex one:
+
+```ruby
+livecheck do
+  url "https://example.org/my-app/latest-mac.yml"
+  regex(/MyApp[._-](\d+(?:\.\d+)+)-mac-ddl-stage-(\d+(?:\.\d+)*)-([0-9a-f]+)\.dmg/i)
+  strategy :electron_builder do |yaml, regex|
+    yaml["files"]&.map do |file|
+      match = file["url"]&.match(regex)
+      next if match.blank?
+
+      "#{match[1]},#{match[2]},#{match[3]}"
```

Comment:
```suggestion
  regex(/MyApp[._-]v?(\d+(?:\.\d+)+)-(\h+)\.dmg/i)
  strategy :electron_builder do |yaml, regex|
    yaml["files"]&.map do |file|
      match = file["url"]&.match(regex)
      next if match.blank?

      "#{match[1]},#{match[2]}"
```

I try to keep the examples as simple as possible while still explaining the desired point. I think we can simplify the regex and `strategy` block here while still explaining the same idea.

This also makes a few tweaks to bring the regex in line with prevailing standards.

-----

For what it's worth, outside of the context of explaining this idea, we also sometimes do something like this in practice to create CSV version values:

```ruby
livecheck do
  url "https://example.org/my-app/latest-mac.yml"
  regex(/MyApp[._-]v?(\d+(?:\.\d+)+-\h+)\.dmg/i)
  strategy :electron_builder do |yaml, regex|
    yaml["files"]&.map do |file|
      match = file["url"]&.match(regex)
      next if match.blank?

      match.tr("-", ",")
   end
  end
end
```

Reply:
should the new CSV example also included?